src/go/rpk/pkg/cli/registry/schema/schema.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package schema
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "schema",
+		Args:  cobra.ExactArgs(0),
+		Short: "Manage your schema registry schemas",
+	}
+	cmd.AddCommand(
+		newCheckCompatibilityCommand(fs, p),
+		newCreateCommand(fs, p),
+		newDeleteCommand(fs, p),
+		newGetCommand(fs, p),
+		newListCommand(fs, p),
+	)
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+type subjectSchema struct {
+	Subject string `json:"subject" yaml:"subject"`
+	Version int    `json:"version" yaml:"version"`
+	ID      int    `json:"ID" yaml:"ID"`
+	Type    string `json:"type" yaml:"type"`
+}
+
+func printSubjectSchemaTable(f config.OutFormatter, single bool, ss ...sr.SubjectSchema) error {
+	var (
+		rows   []subjectSchema
+		tw     *out.TabWriter
+		isText = f.IsText()
+	)
+	if isText {
+		tw = out.NewTable("subject", "version", "id", "type")
+		defer tw.Flush()
+	}
+	for _, s := range ss {
+		r := subjectSchema{
+			Subject: s.Subject,
+			Version: s.Version,
+			ID:      s.ID,
+			Type:    s.Type.String(),
+		}
+		if isText {
+			tw.PrintStructFields(r)
+		} else {
+			rows = append(rows, r)
+		}
+	}
+	// There are commands where a table for text format is fine, but a json
+	// list does not make sense (e.g: schema create)
+	if single && len(rows) == 1 {
+		if _, _, s, err := f.Format(rows[0]); !isText {
+			if err != nil {
+				return fmt.Errorf("unable to print in the required format %q: %v", f.Kind, err)
+			}
+			fmt.Println(s)
+			return nil
+		}
+	}
+	if _, _, s, err := f.Format(rows); !isText {
+		if err != nil {
+			return fmt.Errorf("unable to print in the required format %q: %v", f.Kind, err)
+		}
+		fmt.Println(s)
+		return nil
```

Comment:
I think these two blocks can be changed to something like
```

print := any(rows)
if single {
  print = rows[0]
}
if isText, _, s, err := f.Format(print); !isText {
  ...
  return
}
tw := out.NewTable(...)
defer tw.Flush()
for _, row := range rows {
  ...
}
```
and then get rid of the isText / a bunch of the logic at the top of the func?